src/Hazelcast.Net/Serialization/SerializationService.ReadWrite.cs

```diff
@@ -167,7 +168,7 @@ public bool TryToObject(object dataObj, out object obj, out ToObjectState state)
                 // i.e. have ISerializerAdapter implement TryRead(...)
                 if (serializer is CompactSerializationSerializerAdapter compact)
```

Comment:
It would be great to see a benchmark if polymorphism is faster.

Reply:
yes, that's why I haven't change anything for now - that could be a later optimization task